api/ombpdf/semdom.py

```diff
@@ -0,0 +1,186 @@
+from xml.dom import minidom
+
+from . import semtree
+
+
+class ListInfo:
+    def __init__(self, writer, info):
+        self.writer = writer
+        self.info = info
+        self.item = 1
+
+    def begin_list_item(self):
+        if self.info.is_ordered:
+            attrs = {
+                'emblem': str(self.item),
+                'marker': f'{self.item}.'
+            }
+        else:
+            attrs = {'marker': '\u2022'}
+        self.writer._push_child('listitem', attrs)
+        self.writer._push_child('para')
+        self.writer._push_child('content', elinfo=self)
+        self.item += 1
+
+    def end_list_item(self):
+        self.writer._pop_child('content')
+        self.writer._pop_child('para')
+        self.writer._pop_child('listitem')
+
+
+class DOMWriter(semtree.Writer):
+    def __init__(self):
+        self.muted = True
```

Comment:
It might be helpful to document this flag; I had guessed incorrectly that it skipped over chunks of the document rather than emitting xml comments.

Reply:
Heh, that is actually what it originally did--but I eventually changed the behavior to log comments b/c I found that more helpful for debugging purposes.  I think there might be a better name for what it does now, but I'm not sure what that is, so I guess I'll just add a comment documenting its purpose for now.